### PR TITLE
Week 2 salvador tena

### PR DESCRIPTION
**Pull request by Chava888**
Assessment 2.

### CODE REVIEW

**Comment by Chava888** [comment]
Note to Self. const the functions.

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,10 +22,107 @@ var num2 = 0
...
+    })
+});
+
+describe("divisibleByThree Function", () => {
+
+    it("function takes an input value as an argument and outputs a string if it is or not divisible by three", () => {
+
+        expect(divisibleByThree(num3)).toEqual("-7 is not divisible by three")
+    })
+});
```
Comment (sjproctor) [inline code-challenges.test.js:87]: The syntax looks good, but you can wrap all three expect statements in the same `it` block.

File: code-challenges.test.js [hunk 2]
```diff
@@ -22,10 +22,107 @@ var num2 = 0
...
+
+        expect(divisibleByThree(num1)).toEqual("15 is divisible by three");
+        expect(divisibleByThree(num2)).toEqual("0 is divisible by three");
+        expect(divisibleByThree(num3)).toEqual("-7 is not divisible by three");
+    })
+});
+
+divisibleByThree = (number) => (number % 3 === 0 ? `${number} is divisible by three` : `${number} is not divisible by three`)
+
+
```
Comment (sjproctor) [inline code-challenges.test.js:120]: Oooooooo I see. Nice iteration.

File: code-challenges.test.js [hunk 3]
```diff
@@ -38,8 +135,64 @@ var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction
...
+//Step 3 optimize and make code clear, placed all expect in one it()
+describe("upperCaseArray Function", () => {
+
+    it("function takes an input array as an argument and outputs an array with all capital letters", () => {
+
+        expect(upperCaseArray(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]);
+        expect(upperCaseArray(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]);
+    })
+});
+
```
Comment (sjproctor) [inline code-challenges.test.js:189]: Looks good!

File: coding-concepts.js [hunk 4]
```diff
@@ -8,39 +8,43 @@
...
 
 var cohort = "Echo 2021"
-// console.log(cohort.split(""))
+console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: ["E","c","h","o"," ","2","0","2","1"]
+// b) Verify and explain: Correct. The reason for the output is the split() function splits a string into an array and seperates the string
+// accoring to the input argument. For this case the input argument was "" - no space, therefore separate the string into individual characters.
```
Comment (sjproctor) [inline coding-concepts.js:15]: Nice!

File: coding-concepts.js [hunk 5]
```diff
@@ -8,39 +8,43 @@
...
 // --------------------2) What will this log?
 
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
+
+// a) Your answer: "Hello, LEARN Student!"
+// b) Verify and explain: Wrong, forgot the return so silly on my part. The result was undefined because the function has no return, so the function will not return anything to console.log().
```
Comment (sjproctor) [inline coding-concepts.js:26]: Haha, good explanation.

File: coding-concepts.js [hunk 6]
```diff
@@ -8,39 +8,43 @@
...
 
 var multBy2 = [4, 5, 6, 7, 8].map(value => value * 2)
-// console.log(multBy2)
+console.log(multBy2)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: [8, 10, 12, 14, 16]
+// b) Verify and explain: Correct. The .map() is multiplying every element of the array by a factor of 2. The input function inside .map has an implict arrow function therefore return is not required.
+//Implicit arrow function is ()=>() syntax for one expression, it is implicitly stated the return.
```
Comment (sjproctor) [inline coding-concepts.js:37]: Nice!

File: coding-concepts.js [hunk 7]
```diff
@@ -51,23 +55,30 @@ var myCodingSkills = {
...
-// console.log(learnStudent)
+console.log(learnStudent)
+
+// a) Your answer:{student: "George", cohort : "Delta", year: 2021}
+
+
 
-// a) Your answer:
-// b) Verify and explain:
+// b) Verify and explain: Incorrect. The output was Learn { student: 'George', cohort: 'Delta', year: 2021 },
```
Comment (sjproctor) [inline coding-concepts.js:81]: I'd say you got it. :)

File: interview-questions.md [hunk 8]
```diff
@@ -8,62 +8,79 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 3. React state:
+React state is an object within a React component that contains properties that may change, therefore, are not immutable. React tracks the object state properties and any changes to them. For class components, changes to state properties can be done by using setState(), which merges the changes to the properties. 
+
+For hooks that are used in functional components, state can be any type of datatype that stores a single value. The state value in a hook can be changed by using useState() and a setState variable. In hooks, the state value is replaced and not merged, which is a contrast to a class component.
 
 4. React lifecycle methods:
+React has three lifecycle phases; mounting, updating, and unmounting. Common react lifecycle methods are render(), componentDidMount(), componentDidUpdate(), and componentWillUnmount(). The render() occurs in the mounting and updating phases of React. ComponentDidMount() occurs in the mounting phase of React. ComponentDidUpdate() occurs in the updating react lifecycle. ComponentWillUnmount() occurs in the unmounting phase of react.
```
Comment (sjproctor) [inline interview-questions.md:80]: Nice answer!

**Review by sjproctor — APPROVED** [review]
Really great job, Salvador! Excellent work on the coding concepts and interview questions section. I appreciate the effort you put in. I like the refactor unfolding in the coding section. Can you help me out by having the code you are submitted commented in and the rest commented out? Basically I want to be able to run your file and see passing tests.

**Pull request merged**